### ISSUE TITLE
[DEV] "wget https://xcat.org/files/go-xcat -O - >/tmp/go-xcat" complains "no certificate subject alternative name matches"

### ISSUE THREAD

**Issue by immarvin** [xcat2/xcat-core#2290]
following the usage of go-xcat in http://xcat.org/download.html#xcat-core:

````
c910f04x30v241:~ #        wget https://xcat.org/files/go-xcat -O - >/tmp/go-xcat
--2016-12-07 17:54:19--  https://xcat.org/files/go-xcat
Resolving xcat.org... 166.70.135.166
Connecting to xcat.org|166.70.135.166|:443... connected.
ERROR: no certificate subject alternative name matches
	requested host name xcat.org.
To connect to xcat.org insecurely, use `--no-check-certificate'.
Unable to establish SSL connection.

````

either of the following commands is ok, we need to update the http://xcat.org/download.html#xcat-core page
````
bybc0605:/tmp # wget http://xcat.org/files/go-xcat -O - >/tmp/go-xcat
--2016-12-07 10:36:31--  http://xcat.org/files/go-xcat
Resolving xcat.org (xcat.org)... 166.70.135.166
Connecting to xcat.org (xcat.org)|166.70.135.166|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 39750 (39K)
Saving to: 'STDOUT'

100%[========================================================================================>] 39,750      --.-K/s   in 0.06s

2016-12-07 10:36:31 (696 KB/s) - written to stdout [39750/39750]

bybc0605:/tmp # wget https://xcat.org/files/go-xcat --no-check-certificate -O - >/tmp/go-xcat
--2016-12-07 10:36:42--  https://xcat.org/files/go-xcat
Resolving xcat.org (xcat.org)... 166.70.135.166
Connecting to xcat.org (xcat.org)|166.70.135.166|:443... connected.
WARNING: cannot verify xcat.org's certificate, issued by '/C=US/O=Let\'s Encrypt/CN=Let\'s Encrypt Authority X3':
  Unable to locally verify the issuer's authority.
HTTP request sent, awaiting response... 200 OK
Length: 39750 (39K)
Saving to: 'STDOUT'

100%[========================================================================================>] 39,750      --.-K/s   in 0.06s

2016-12-07 10:36:43 (604 KB/s) - written to stdout [39750/39750]

````

**Comment by whowutwut** [comment]
@immarvin  I've seen this issue dating back to 5/5/16 when we switched to a SSL certificate from Let's Encrypt and had talked to @daniceexi  about this.  But didn't investigate further.  I was almost thinking we should just not use https.    But doing more investigation on this today, it seems like only certain OS experience this. 

 RHEL 7.3 **DOES NOT** have a problem 
```
[root@fs4 vhu]# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.3 (Maipo)

[root@fs4 vhu]# wget https://xcat.org/files/go-xcat -O - > ./go-xcat
--2016-12-07 10:27:27--  https://xcat.org/files/go-xcat
Resolving xcat.org (xcat.org)... 166.70.135.166
Connecting to xcat.org (xcat.org)|166.70.135.166|:443... connected.

HTTP request sent, awaiting response... 200 OK
Length: 39750 (39K)
Saving to: ‘STDOUT’

100%[==================================================================================================>] 39,750      --.-K/s   in 0.06s

2016-12-07 10:27:27 (674 KB/s) - written to stdout [39750/39750]
```

SLES 12.x **DOES** have a problem 
```
bybc0605:~ # cat /etc/SuSE-release
SUSE Linux Enterprise Server 12 (x86_64)
VERSION = 12
PATCHLEVEL = 1
# This file is deprecated and will be removed in a future service pack or release.
# Please check /etc/os-release for details about this release.

bybc0605:~ # wget https://xcat.org/files/go-xcat -O - >/tmp/go-xcat
--2016-12-07 13:21:43--  https://xcat.org/files/go-xcat
Resolving xcat.org (xcat.org)... 166.70.135.166
Connecting to xcat.org (xcat.org)|166.70.135.166|:443... connected.
ERROR: cannot verify xcat.org's certificate, issued by '/C=US/O=Let\'s Encrypt/CN=Let\'s Encrypt Authority X3':
  Unable to locally verify the issuer's authority.
To connect to xcat.org insecurely, use `--no-check-certificate'.
```

I ran `xcat.org` through some online SSL checkers (https://www.ssllabs.com/) and also https://whatsmychaincert.com to make sure that the certificate chain is OK,  It all looks good...

Looking further, I think it's caused by the ca-certificates.  The `ca-certificates` package is provided by  the OS and SLES 12 is a 2014 version:
```
c910f04x42vm11:~ # rpm -qa | grep ca-
ca-certificates-1_201403302107-6.2.noarch
yast2-ca-management-3.1.7-1.126.noarch
ca-certificates-mozilla-2.2-7.1.noarch
```
...where as the one provided by RHEL 7.3 is a 2015 version.: 
```
[root@fs4 ~]# rpm -qa | grep ca-cer
ca-certificates-2015.2.6-73.el7.noarch
```
This could be the cause? 

Following this thread: https://community.letsencrypt.org/t/wget-cannot-verify-certificate/17651/6
The last comment is we can build the certificates manually and pass it to the `wget` command.  

> if that's not practical, you can grab just the DST Root X3 and ISRG Root X1 certificates from https://letsencrypt.org/certificates/27 and make your own ca-certificates file. Then pass the --ca-certificates flag to wget.

 Do you have experience doing this? 

If this is the reason, we can provide some additional steps in the documentation to tell the user how to generate the cross signing certificate and provide it to wget

I don't think this needs to be resolved in 2.13 release..

**Comment by immarvin** [comment]
fixed in  https://github.com/xcat2/xcat.org/pull/41 , https://github.com/xcat2/xcat-extensions/pull/3
